src/app/dashboard/roles/page.tsx

```
@@ -84,21 +86,51 @@ export default function RolesPage() {
+              try {
+                await deleteRole(storeId, id);
+                setRoles((prev) => prev.filter((role) => role.id !== id));
+                setFilteredRoles((prev) => prev.filter((role) => role.id !== id));
+                toast.success('Role deleted successfully');
+              } catch (err) {
+                console.error('Error deleting role:', err);
+                toast.error('Failed to delete role');
+              }
+            }}
+            className="bg-red-500 hover:bg-red-600 text-white text-xs px-3 py-1 rounded"
+          >
```

Biome (2.1.2)

[error] Provide an explicit type prop for the button element.

The default type of a button is submit, which causes the submission of a form when placed inside a `form` element. This is likely not the behaviour that you want inside a React application.
Allowed button types are: submit, button or reset

(lint/a11y/useButtonType)